shared/dataflow/codeql/dataflow/internal/DataFlowImpl.qll

```diff
@@ -593,11 +594,30 @@ module MakeImpl<LocationSig Location, InputSig<Location> Lang> {
         )
       }
 
+      pragma[nomagic]
+      private predicate fwdFlowInParam(DataFlowCall call, ParamNodeEx p, Cc cc) {
+        fwdFlowIn(call, _, cc, p)
+      }
```

Comment:
The columns `call` and `p` ought to functionally determine `arg` on almost all cases, so I think it's better to inline this projection rather than materialising additional tuples.